### PR TITLE
Add support for Unix Domain Socket (UDS) configuration via `DD_TRACE_AGENT_URL`

### LINKED ISSUE #1967: Missing support for Unix Domain Socket configuration via `DD_TRACE_AGENT_URL`
Spotted that we're missing support for specifying a UDS path via `DD_TRACE_AGENT_URL`, which other Datadog libraries support.

### PR DESCRIPTION
**Pull request by ivoanjo**
**What does this PR do?**:

This PR extends the `AgentSettingsResolver` to allow the `DD_TRACE_AGENT_URL` to contain a Unix Domain Socket (UDS) configuration string, for instance `unix:///some/path/to/the/apm.socket`.

It also adds a warning if there's configuration for both http and unix domain socket (and in that case it will prioritize the http variant).

**Motivation**:

This was missing for feature parity with other Datadog client libraries, and it was annoying a lot of customers in #1967 (thanks for the patience, folks!).

**Additional Notes**:

The change is quite small because dd-trace-rb already supported using unix domain sockets -- you just couldn't configure them via this environment variable. Thus, the only thing that changed was parsing it correctly.

**How to test the change?**:

Configure an agent with a uds socket and report data through it.

Fixes #1967

### CODE REVIEW

**Review by GustavoCaso — APPROVED** [review]
LGTM

**Review by marcotc — APPROVED** [review]

**Comment by ivoanjo** [comment]
Hey! I found a tiny tiny bug that wouldn't even impact customers at this point but I want to fix it and add test coverage for it before merging this PR, so I've marked this as do not merge :)

**Comment by ivoanjo** [comment]
Tiny bug fixed in https://github.com/DataDog/dd-trace-rb/pull/2806/commits/e7680ed15e9964362cd6270e7520659ad4a630b7

**Review by marcotc — APPROVED** [review]

**Comment by ivoanjo** [comment]
Thanks y'all for the reviews, merging away!

**Pull request merged**
